### ISSUE TITLE
[Initiative] Add support for single-sourced license level information

### ISSUE THREAD

**Issue by bmorelli25** [elastic/docs-builder#101]
### Summary

https://www.elastic.co/subscriptions is the source of truth for feature licensing information. We do not list specific licensing information in the docs as it inevitably changes over time.

In V3, we should explore the ability to single-source feature licensing information. For example, if we had a single yaml or json file that the product team could add licensing information to, we could potentially show specific license levels in the docs. IMO, this would be a better user experience than forcing folks to go to a different page to determine if the feature they want to use is supported by their current subscription.

Alternatively, from a Slack convo:

> It makes sense to not scatter subscriptions all over the docs. I just had a hard time figuring out which one was related to this feature. Maybe all features in the subscriptions page should have anchors that doc pages can link to directly

**Comment by lcawl** [comment]
My two cents would be that we'd still want these details to point to the website as the source of truth in the case folks forget to keep them in sync.

**Comment by bmorelli25** [comment]
I don't know what I'm envisioning yet tbh. But I do think the ideal user experience would be to not have to leave the page to see the subscription level of the feature you're reading about. I'm also thinking that when we eventually allow users to log into the docs, we'd want to be able to call out that the feature they're reading about is or isn't supported by their current subscription level.

**Comment by cotti** [comment]
## AI triage

_Automated investigation against the current codebase. A human still needs to confirm before any code change._

### Original concern

Stop scattering subscription/license information across the docs by keeping it in one source-of-truth file that pages can reference. Better UX than sending readers to `https://www.elastic.co/subscriptions` every time.

### Findings

- The `applies_to` model already covers several availability axes but explicitly **does not** cover subscription/license tier. See `src/Elastic.Documentation/AppliesTo/ApplicableTo.cs:40-69` — the rich model has `Stack`, `Deployment` (self/ece/eck/ess), `Serverless` (es/obs/sec), and `Product`/`ProductApplicability`, but no `License` / `Subscription` / `Tier` property. `Lifecycle` (`ProductLifecycle.GenerallyAvailable`, `Preview`, `Beta`, etc., at lines 134-146) is about maturity, not entitlement.
- The wire-format `AppliesToEntry` indexed in Elasticsearch (`ToAppliesTo()` at lines 75-120) reflects the same omission: no subscription dimension. Adding one would require schema work in `AppliesTo/`, `Configuration/`, parser support, renderer support, and a corresponding migration of how `applies_to` is documented.
- No existing single-source file (YAML/JSON) of `feature → tier` exists in the repo or in `config/`. There is no precedent in `docs-builder` for the "shared registry of facts" pattern this initiative would need.
- There is meaningful prior art: `elastic/docs#476` (linked in the original issue) proposed essentially the same thing in the asciidoc era. It was closed in 2019 with `"we've been told explicitly that we do not want to do this."` Engineering leadership at the time rejected the model. That decision needs to be revisited deliberately, not implicitly.
- @lcawl raised in-thread concerns that still apply: (1) the website remains the canonical source — anything in the docs risks drifting; (2) page-level granularity is awkward when one page covers multiple features; (3) inline-everywhere is admonition-noise territory.
- The issue carries the `After 9.0` label, signaling it's explicitly deferred.

### Verdict

**Needs stakeholder input**

The technical shape is tractable (extend `ApplicableTo` with a subscription axis, add a registry under `config/`, render a badge component), but every meaningful aspect — should we do this at all, what's the source-of-truth contract with the subscriptions page, how do we surface it without becoming noise — was previously decided against and has not been re-decided. Implementation without that re-decision risks shipping something that gets reverted again.

### Proposal

Open questions, grouped by audience:

**Product**
- Is the 2019 "no" still in effect, or has the product position changed? Confirm explicit re-decision before we plan implementation. (If no re-decision, this issue should be closed or kept frozen.)
- Authoritative source: stays on `elastic.co/subscriptions`, or does a docs-side registry become the source of truth? If the website stays canonical, define the contract for staying in sync (manual review cadence, automated sync, scheduled audit?).
- Cloud vs self-managed: tier names differ between [the main subscriptions page](https://www.elastic.co/subscriptions) and [the cloud one](https://www.elastic.co/cloud/elasticsearch-service/subscriptions). Pick one vocabulary or commit to expressing both.
- Is the "show entitlement based on logged-in user" piece part of this initiative, or a separate follow-up? It changes the rendering surface dramatically.

**UX**
- Surface treatment: per-page metadata (like `applies_to` today), prerequisite section, inline badge near the feature name, hover/popover, or some mix? The `applies_to` model already has a render slot — would license tier reuse the same chip strip or be visually separate?
- Granularity: if one page covers multiple features at different tiers, do we render multiple badges or fall back to the most-restrictive?
- Reader who isn't logged in vs reader who is: do we show their match/mismatch, or just always show the requirement?

**Writer**
- Authoring syntax preference: extend `applies_to:` with a new key (e.g. `subscription: gold`), or a separate `license:` frontmatter field, or a directive? Each has migration implications for the corpus.
- One-time migration: who owns annotating existing pages once syntax lands?

**Engineering**
- If we add a subscription axis to `ApplicableTo`, it cascades to: YAML schema, JSON converter (`ApplicableToJsonConverter`), search-index wire format (`AppliesToEntry`), renderer, link index, and `tests/Elastic.Markdown.Tests/AppliesTo/`. Scope is non-trivial.
- Where does the `feature → tier` registry live, and how does it version against Stack versions (e.g. a feature that was Gold in 8.x but Basic in 9.x)?

### Notes

The "After 9.0" label and the dormant 2019 precedent suggest the right next step is a short product/eng sync to either revive or formally close this initiative, **before** any code is proposed here. Suggested follow-up: a focused product issue (not necessarily in this repo) asking the explicit question "Are we doing license-tier metadata in docs in 2026, yes or no?"
